File: lib/src/native/video_recorder/record_event.dart
import 'dart:convert';

class RecordEvent {
  RecordEvent({
    required this.type,
    this.data,
  });

  factory RecordEvent.fromMap(Map<Object?, Object?> map) =>
      RecordEvent.fromJson(jsonDecode(jsonEncode(map)));

  factory RecordEvent.fromJson(Map<String, dynamic> json) {
    return RecordEvent(
      type: RecordEventType.fromString(json['type']),
      data: json['data'] as Map<String, dynamic>?,
    );
  }

  final RecordEventType type;
  final Map<String, dynamic>? data;
}

enum RecordEventType {
  idle,
  starting,
  recording,
  stop,
  result,
  error,
  ;

  static RecordEventType fromString(String value) =>
      RecordEventType.values.firstWhere((element) => element.name == value);
}

File: android/src/main/java/com/cloudwebrtc/webrtc/videoRecorder/RecordEvent.kt
package com.cloudwebrtc.webrtc.videoRecorder

public data class RecordEvent (
    val type: RecordEventType,
    val data: Map<String, Any?>?
) {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "type" to type.name,
            "data" to data
        )
    }
}
